### DISCUSSION TITLE
74. 데이터베이스 정규화는 무엇이고, 왜 사용하는지 설명해 주세요.

### DISCUSSION THREAD

**Discussion by cutehumanS2** [jewelry-box/cs-interview#76] [General]
데이터베이스 정규화는 무엇이고, 왜 사용하는지 설명해 주세요.

**Comment by eun-byeol** [comment]
정규화는 관계형 DB에서 `데이터 중복을 최소화`하고 `데이터 일관성을 유지`하기 위해 데이터 구조를 조직화하는 프로세스 입니다.
데이터 중복으로 인한 `이상 현상(anomaly)을 제거`하기 위해 사용합니다.
1NF, 2NF, 3NF, BCNF, 4NF, 5NF, 6NF 정규형이 있습니다. 앞 단계를 만족해야 다음 단계를 진행할 수 있습니다.

**Comment by eun-byeol** [comment]
## 정규화 원칙
1. 정보의 무손실 표현 : 하나의 스키마를 다른 스키마로 변환할 때 정보의 손실이 있어선 안 된다.
2. 분리의 원칙 : 하나의 독립된 관계성은 하나의 독립된 릴레이션으로 분리시켜 표현해야 한다.
3. 데이터의 중복성이 감소되어야 한다.

## 이상 현상
1. `삽입 이상` : 원하지 않는 데이터 삽입되거나 삽입되지 않는 문제
2. `삭제 이상` : 원하지 않는 정보 손실이 발생하는 문제. 하나의 데이터만 삭제하고 싶은데, 그 데이터에 포함된 튜플 전체가 삭제되는 경우
3. `수정 이상`(갱신 이상) : 튜플의 일부만 갱신되어 일관성이 없어져 발생하는 문제
